### PR TITLE
Clean up handling of Uri

### PR DESCRIPTION
**Pull request by olafurpg**
This PR is a refactoring of how we handle URIs. Most importantly, we do no more `String -> URI -> AbsolutePath - URI -> String` roundtrips. While working on this PR observed that `Path.toUri.toString` produces `file:///path` instead of `file:/path` which lsp expects. The logic to produce `s"file:$path"` was spread all over the place and made it really tricky to debug when things went wrong. Now we only manipulate `Uri` which is a plain wrapper around a String.

### CODE REVIEW

**Comment by laughedelic** [comment]
I'll review the changes tomorrow, just a note about 

> file:///path instead of file:/path which lsp expects

Both formats are correct URIs and LSP doesn't enforce any particular format. The latter one is a more recent standard. Check https://github.com/sbt/sbt/issues/3801, Eugene has a good explanation and references to the standards there. Using the `file:///path` is better in the sense that it should be supported by older implementations. Does it cause any problems in VSCode? I thought that it uses exactly this format and `file:/path` comes from the Java `file.toURI` method.

**Comment by olafurpg** [comment]
Interesting, the reason it matters to us is that we persist these URIs as strings and look them up as strings so we'll have to enforce either one. I have no preference which one, with everything under `Uri` we can decide to use `file:///` instead.

File: metaserver/src/main/scala/scala/meta/languageserver/Uri.scala [hunk 1]
```diff
@@ -1,15 +1,46 @@
...
+import langserver.types.VersionedTextDocumentIdentifier
+import org.langmeta.inputs.Input
 import org.langmeta.io.AbsolutePath
 
-/** Extractor for converting stringy vscode uris into absolute paths */
+/**
+ * Wrapper for a string representing an LSP-compliant URI.
+ *
+ * This is not a java.net.URI because
+ * - URI a lot of methods that return null and we don't use anyways
```
Comment (gabro) [inline metaserver/src/main/scala/scala/meta/languageserver/Uri.scala]: missing verb after `URI`?

**Comment by olafurpg** [comment]
This makes no sense 😿 
```scala
@ URI.create("file:///blah") == URI.create("file:/blah")
res6: Boolean = true

@ URI.create("file://blah") == URI.create("file:/blah")
res7: Boolean = false

@ URI.create("file:/blah") == URI.create("file:/blah")
res8: Boolean = true
```

I didn't know about that @laughedelic interesting. I updated the implementation to enforce `file:///path`, now that everything is under `Uri` we can easily adapt if we prefer to work with `java.net.URI` etc

**Comment by laughedelic** [comment]
Why doesn't it make sense? `file://blah` is a relative path, isn't it? and `file:/blah` is an absolute path. So I think it makes sense.. I think the simple trick to transform such string URI (with either format) to path is to do `.stripPrefix("file://")`

**Pull request merged**